aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -124,8 +125,8 @@ private LockManager initializeLockManager(Map<String, String> properties) {
 
   private FileIO initializeFileIO(Map<String, String> properties) {
     String fileIOImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
-    if (fileIOImpl == null) {
-      FileIO io = new S3FileIO();
+    if (fileIOImpl == null || fileIOImpl.equals(S3_FILE_IO_IMPL)) {
+      FileIO io = new S3FileIO(properties);
```

Comment:
I don't think you need to do this, properties will be passed in during `io.initialize(properties);` at next line anyway